rust/benchmarks/src/bin/tpch.rs

```diff
@@ -1205,3 +1206,406 @@ fn get_schema(table: &str) -> Schema {
+
+    #[tokio::test]
+    async fn q1() -> Result<()> {
+        verify_query(1).await
+    }
+
+    #[tokio::test]
+    async fn q2() -> Result<()> {
+        verify_query(2).await
+    }
+
+    #[tokio::test]
+    async fn q3() -> Result<()> {
+        verify_query(3).await
+    }
+
+    #[tokio::test]
+    async fn q4() -> Result<()> {
+        verify_query(4).await
+    }
+
+    #[tokio::test]
+    async fn q5() -> Result<()> {
+        verify_query(5).await
+    }
+
+    #[tokio::test]
+    async fn q6() -> Result<()> {
+        verify_query(6).await
+    }
+
+    #[tokio::test]
+    async fn q7() -> Result<()> {
+        verify_query(7).await
+    }
+
+    #[tokio::test]
+    async fn q8() -> Result<()> {
+        verify_query(8).await
+    }
+
+    #[tokio::test]
+    async fn q9() -> Result<()> {
+        verify_query(9).await
+    }
+
+    #[tokio::test]
+    async fn q10() -> Result<()> {
+        verify_query(10).await
+    }
+
+    #[tokio::test]
+    async fn q11() -> Result<()> {
+        verify_query(11).await
+    }
+
+    #[tokio::test]
+    async fn q12() -> Result<()> {
+        verify_query(12).await
+    }
+
+    #[tokio::test]
+    async fn q13() -> Result<()> {
+        verify_query(13).await
+    }
+
+    #[tokio::test]
+    async fn q14() -> Result<()> {
+        verify_query(14).await
+    }
+
+    #[tokio::test]
+    async fn q15() -> Result<()> {
+        verify_query(15).await
+    }
+
+    #[tokio::test]
+    async fn q16() -> Result<()> {
+        verify_query(16).await
+    }
+
+    #[tokio::test]
+    async fn q17() -> Result<()> {
+        verify_query(17).await
+    }
+
+    #[tokio::test]
+    async fn q18() -> Result<()> {
+        verify_query(18).await
+    }
+
+    #[tokio::test]
+    async fn q19() -> Result<()> {
+        verify_query(19).await
+    }
+
+    #[tokio::test]
+    async fn q20() -> Result<()> {
+        verify_query(20).await
+    }
+
+    #[tokio::test]
+    async fn q21() -> Result<()> {
+        verify_query(21).await
+    }
+
+    #[tokio::test]
+    async fn q22() -> Result<()> {
+        verify_query(22).await
+    }
+
+    /// Specialised String representation
+    fn col_str(column: &ArrayRef, row_index: usize) -> String {
+        if column.is_null(row_index) {
+            return "NULL".to_string();
+        }
+
+        // Special case ListArray as there is no pretty print support for it yet
+        if let DataType::FixedSizeList(_, n) = column.data_type() {
+            let array = column
+                .as_any()
+                .downcast_ref::<FixedSizeListArray>()
+                .unwrap()
+                .value(row_index);
+
+            let mut r = Vec::with_capacity(*n as usize);
+            for i in 0..*n {
+                r.push(col_str(&array, i as usize));
+            }
+            return format!("[{}]", r.join(","));
+        }
+
+        array_value_to_string(column, row_index)
+            .ok()
+            .unwrap_or_else(|| "???".to_string())
```

Comment:
Better to have either a good error message or do normal unwrap if it doesn't occur in test code.

Reply:
Thanks for the review. To be honest I had copied that from https://github.com/apache/arrow/blob/master/rust/datafusion/tests/sql.rs#L1387 but i have fixed it to panic in the test.